### PR TITLE
secvarctl-tests.py: Adjust to upstream changes

### PR DESCRIPTION
**Pull request by nasastry**
Adjust to upstream changes

### CODE REVIEW

**Comment by nasastry** [comment]
This PR is depending on https://github.com/avocado-framework/avocado/pull/5910

**Comment by nasastry** [comment]
SLES run:
```
# avocado run --max-parallel-tasks=1 secvarctl-tests.py -m secvarctl-tests.py.data/secvarctl.yaml
JOB ID     : d17ae43a4f90263cd941efab61ba050d4722059f
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2024-04-18T14.39-d17ae43/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: STARTED
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (16.89 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: STARTED
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: PASS (18.44 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2024-04-18T14.39-d17ae43/results.html
JOB TIME   : 58.84 s
```

RHEL run: In RHEL Operating System secvarctl rpm is not yet there so failure is expected.
```
# avocado run --max-parallel-tasks=1 secvarctl-tests.py -m secvarctl-tests.py.data/secvarctl.yaml
JOB ID     : 833cb009870b6431da97aa1c2c09904876e425e1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2024-04-18T04.11-833cb00/job.log
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: STARTED
 (1/2) secvarctl-tests.py:secvarctl.test;run-run_type-upstream-131e: PASS (27.49 s)
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: STARTED
 (2/2) secvarctl-tests.py:secvarctl.test;run-run_type-distro-17a2: ERROR: secvarctl source install failed. (1.26 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2024-04-18T04.11-833cb00/results.html
JOB TIME   : 47.06 s
```

**Comment by nasastry** [comment]
> This PR is depending on [avocado-framework/avocado#5910](https://github.com/avocado-framework/avocado/pull/5910)

The above avocado PR got merged.

File: security/secvarctl-tests.py [hunk 1]
```diff
@@ -31,35 +31,46 @@ def setUp(self):
...
             if not self.srcdir:
                 self.fail("secvarctl source install failed.")
-            self.srcdir = os.path.join(self.srcdir, 'test')
-        os.chdir(self.srcdir)
+        self.build_dir = os.path.join(self.srcdir, "build")
+        os.mkdir(self.build_dir)
+        os.chdir(self.build_dir)
+        rc = process.system("cmake ../")
+        if rc:
+            self.cancel("Can't run 'cmake'")
```
Comment (PraveenPenguin) [inline security/secvarctl-tests.py:62]: Can have some more meaningful message , one suggestion is 
`test need to cancel as running cmake has issue , please check environment `

File: security/secvarctl-tests.py [hunk 2]
```diff
@@ -31,35 +31,46 @@ def setUp(self):
...
     def test(self):
         '''
         Running tests from secvarctl
         '''
         count = 0
-        output = build.run_make(self.srcdir, extra_args="OPENSSL=1",
+        output = build.run_make(self.build_dir,
+                                process_kwargs={"ignore_status": True})
+        if output.exit_status:
+            self.cancel("secvarctl 'make' failed.")
```
Comment (PraveenPenguin) [inline security/secvarctl-tests.py:72]: Here also please enhance message

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**